app/assets/javascripts/application.js

```diff
@@ -667,23 +667,32 @@ function renderMorpheus(dataPath, annotPath, selectedAnnot, selectedAnnotType, t
 }
 
 // toggles visibility and disabled status of file upload and fastq url fields
-function toggleFastqFields(target) {
+function toggleFastqFields(target, state) {
     var selector = $("#" + target);
     var fileField = selector.find('.upload-field');
-    $(fileField).toggleClass('hidden');
     var fastqField = selector.find('.fastq-field');
-    $(fastqField).toggleClass('hidden');
-    // toggle disabled status by returning inverse of current state
-    $(fastqField).find('input').attr('disabled', !$(fastqField).find('input').is('[disabled=disabled]'));
-    // set human data attr to true
     var humanData = $(fastqField).find('input[type=hidden]');
-    $(humanData).val($(humanData).val() === 'true' ? 'false' : 'true' );
-    // enable name field & update button to allow saving
     var saveBtn = selector.find('.save-study-file');
-    $(saveBtn).attr('disabled', !$(saveBtn).is('[disabled=disabled]'));
     var nameField = selector.find('.filename');
-    $(nameField).attr('readonly', !$(nameField).is('[readonly=readonly]'));
-    $(nameField).attr('placeholder', '');
+    if (state === 'true') {
```

Comment:
It looks like `state` is intended to represent a Boolean.  If its only values should be `'true'` or `'false'`, consider casting it as a Boolean before passing it into this function -- even it's parsed from a DOM value, as I imagine it is.